pkg/awsutils/awsutils.go

```diff
@@ -731,22 +731,33 @@ func (cache *EC2InstanceMetadataCache) FreeENI(eniName string) error {
 		log.Errorf("Failed to retrieve ENI %s attachment id: %v", eniName, err)
 		return errors.Wrap(err, "FreeENI: failed to retrieve ENI's attachment id")
 	}
-
 	log.Debugf("Found ENI %s attachment id: %s ", eniName, aws.StringValue(attachID))
 
-	// Detach it first
 	detachInput := &ec2.DetachNetworkInterfaceInput{
 		AttachmentId: attachID,
-		Force:        aws.Bool(true),
 	}
 
-	start := time.Now()
-	_, err = cache.ec2SVC.DetachNetworkInterface(detachInput)
-	awsAPILatency.WithLabelValues("DetachNetworkInterface", fmt.Sprint(err != nil)).Observe(msSince(start))
-	if err != nil {
-		awsAPIErrInc("DetachNetworkInterface", err)
-		log.Errorf("Failed to detach ENI %s %v", eniName, err)
-		return errors.Wrap(err, "FreeENI: failed to detach ENI from instance")
+	// Retry detaching the ENI from the instance
+	var retry int
+	for retry := 0; retry < maxENIDeleteRetries; retry++ {
+		start := time.Now()
+		_, err = cache.ec2SVC.DetachNetworkInterface(detachInput)
+		awsAPILatency.WithLabelValues("DetachNetworkInterface", fmt.Sprint(err != nil)).Observe(msSince(start))
+		if err != nil {
+			awsAPIErrInc("DetachNetworkInterface", err)
+			log.Errorf("Failed to detach ENI %s %v", eniName, err)
+		}
+		if err == nil {
+			log.Infof("Successfully detached ENI: %s", eniName)
+			break
+		}
+
+		log.Debugf("Not able to detach ENI yet (attempt %d/%d): %v ", retry, maxENIDeleteRetries, err)
+		time.Sleep(retryDeleteENIInternal)
+	}
+
+	if retry >= maxENIDeleteRetries {
```

Comment:
Retry can never be more than max.
Also, this will give a failure even if the for loop passes on the last retry. 
Some options are a bool, or turning this into a function.
Another is to move 759 (and make it ==) to 749.

Reply:
Ah, made it `==`, but then the loop needs to start at 1. PR updated